src/shared/directives/substitutor/services/placeholder.service.ts

```diff
@@ -18,58 +19,88 @@ ___
       ) { }
 
     resolvePlaceholders(pageFormFields, stringToResolve): string {
```

Comment:
Function way too long and complicated to read. Could you please break into smaller ones?